server/types/type.go

```diff
@@ -519,34 +524,10 @@ func (t *DoltgresType) MaxCharacterLength() int64 {
 
 // MaxSerializedWidth implements the types.ExtendedType interface.
 func (t *DoltgresType) MaxSerializedWidth() types.ExtendedTypeSerializedWidth {
-	// TODO: need better way to get accurate result
-	switch t.TypCategory {
-	case TypeCategory_ArrayTypes:
-		return types.ExtendedTypeSerializedWidth_Unbounded
-	case TypeCategory_BooleanTypes:
-		return types.ExtendedTypeSerializedWidth_64K
-	case TypeCategory_CompositeTypes, TypeCategory_EnumTypes, TypeCategory_GeometricTypes, TypeCategory_NetworkAddressTypes,
-		TypeCategory_RangeTypes, TypeCategory_PseudoTypes, TypeCategory_UserDefinedTypes, TypeCategory_BitStringTypes,
-		TypeCategory_InternalUseTypes:
-		return types.ExtendedTypeSerializedWidth_Unbounded
-	case TypeCategory_DateTimeTypes:
-		return types.ExtendedTypeSerializedWidth_64K
-	case TypeCategory_NumericTypes:
-		return types.ExtendedTypeSerializedWidth_64K
-	case TypeCategory_StringTypes, TypeCategory_UnknownTypes:
-		if t.ID == VarChar.ID {
-			l := t.Length()
-			if l != StringUnbounded && l <= stringInline {
-				return types.ExtendedTypeSerializedWidth_64K
-			}
-		}
-		return types.ExtendedTypeSerializedWidth_Unbounded
-	case TypeCategory_TimespanTypes:
-		return types.ExtendedTypeSerializedWidth_64K
-	default:
-		// shouldn't happen
+	if t.TypLength < 0 {
```

Comment:
This change seems fine to me. It looks like all cases where the `length == -1` corresponds to unbounded types.